infra/wca_on_rails/production/pipeline.tf

```diff
@@ -0,0 +1,313 @@
+  policy = data.aws_iam_policy_document.codepipeline_policy.json
+}
+
+resource "aws_codebuild_project" "build" {
+  name         = "${var.name_prefix}-build"
+  service_role = aws_iam_role.codepipeline_role.arn
+
+  artifacts {
+    type = "CODEPIPELINE"
+  }
+
+  environment {
+    compute_type                = "BUILD_GENERAL1_SMALL"
+    type                        = "LINUX_CONTAINER"
+    image                       = "aws/codebuild/standard:6.0"
+    image_pull_credentials_type = "CODEBUILD"
+  }
+
+  source {
+    type = "CODEPIPELINE"
+    buildspec = templatefile("../templates/buildspec_build.yml.tftpl", {
+      container_name         = "rails-main"
+      container_port         = 3000
+      task_definition        = aws_ecs_task_definition.this.arn
+      capacity_provider_name = var.shared.m6i_capacity_provider.name
+    })
+  }
+}
+
+data "aws_iam_policy_document" "codedeploy_assume_role_policy" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["codedeploy.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "codedeploy_role" {
+  name               = "${var.name_prefix}-codedeploy-role"
+  assume_role_policy = data.aws_iam_policy_document.codedeploy_assume_role_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "codedeploy_role_attachment" {
+  role       = aws_iam_role.codedeploy_role.name
+  policy_arn = "arn:aws:iam::aws:policy/AWSCodeDeployRoleForECS"
+}
+
+resource "aws_codedeploy_app" "this" {
+  name             = var.name_prefix
+  compute_platform = "ECS"
+}
+
+resource "aws_codedeploy_deployment_group" "this" {
+  app_name               = aws_codedeploy_app.this.name
+  deployment_group_name  = var.name_prefix
+  deployment_config_name = "CodeDeployDefault.ECSAllAtOnce"
+  service_role_arn       = aws_iam_role.codedeploy_role.arn
+
+  blue_green_deployment_config {
+    deployment_ready_option {
+      action_on_timeout = "CONTINUE_DEPLOYMENT"
+    }
+
+    terminate_blue_instances_on_deployment_success {
+      action                           = "TERMINATE"
+      termination_wait_time_in_minutes = 5
+    }
+  }
+
+  deployment_style {
+    deployment_type   = "BLUE_GREEN"
+    deployment_option = "WITH_TRAFFIC_CONTROL"
+  }
+
+  auto_rollback_configuration {
+    enabled = true
+    events  = ["DEPLOYMENT_FAILURE"]
+  }
+
+  ecs_service {
+    cluster_name = var.shared.ecs_cluster.name
+    service_name = aws_ecs_service.this.name
+  }
+
+  load_balancer_info {
+    target_group_pair_info {
+      prod_traffic_route {
+        listener_arns = [var.shared.https_listener.arn]
+      }
+
+      target_group {
+        name = var.shared.rails-blue-green[0].name
+      }
+
+      target_group {
+        name = var.shared.rails-blue-green[1].name
+      }
+    }
+  }
+}
+
+resource "aws_s3_bucket" "this" {
+  bucket        = var.name_prefix
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "this" {
```

Comment:
verify whether we need this

Reply:
The only allowed values for type here
artifact_store {
    type     = "S3"
    location = aws_s3_bucket.this.bucket
  }
is S3, so we do need this. Not sure why AWS even gives us an option